app/core/scheduler/scheduler.py

```diff
@@ -2,40 +2,51 @@
 # For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
 
 import logging
-from typing import NoReturn
+import os
+import signal
 
 from astropy.time import Time
 
-from app.config import config
-from app.core.scheduler.modes import SimulationMode, ValidationMode, OperationMode, SchedulerMode
-
+from app.core.programprovider.ocs import read_ocs_zipfile, OcsProgramProvider
+from app.core.builder import SchedulerBuilder, Blueprints
+from app.db.planmanager import PlanManager
+from definitions import ROOT_DIR
 
 class Scheduler:
-
     def __init__(self, start_time: Time, end_time: Time):
         self.start_time = start_time
         self.end_time = end_time
-        self._mode = None
-    
-    def set_mode(self, mode: SchedulerMode) -> NoReturn:
-        self._mode = mode
-
+ 
     def __call__(self):
```

Comment:
I think this part is the more important one. This is our main recipe. The way we schedule. And should be as inmutable as possible (ofc it might change in the future) I think is good to have a clean main flow and the branches and other options in a separate way. For us this is the basic way the Process Manager handles a schedule, other things should be under the hood.